File: neo/src/dal/models/MixtapePreview.ts
import * as t from 'io-ts';
import { date as dateType } from 'io-ts-types/lib/date';

export const codec = t.type({
  id: t.number,
  title: t.string,
  slug: t.string,
  userId: t.number,
  createdAt: dateType,
  publishedAt: t.union([dateType, t.null]),
  songCount: t.string,
  authorName: t.string,
});

export type Model = t.TypeOf<typeof codec>;

File: rhiannon/src/main/kotlin/club/jambuds/model/MixtapePreview.kt
package club.jambuds.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant
import org.jdbi.v3.core.mapper.Nested

data class MixtapePreview(
    val id: Int,
    @JsonIgnore val createdAt: Instant?,
    @JsonIgnore val userId: Int?,
    val title: String,
    val slug: String,
    val publishedAt: Instant?,
    val songCount: Int,
    val authorName: String,
    @Nested("meta") val meta: ItemMeta
)
